Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/calculation/PresenceTracingRiskMapper.kt

```diff
@@ -46,11 +59,15 @@ class PresenceTracingRiskMapper @Inject constructor(
         getRiskCalculationParameters()?.normalizedTimePerCheckInToRiskLevelMapping
 
     private suspend fun getRiskCalculationParameters(): PresenceTracingRiskCalculationParamContainer? {
-        if (presenceTracingRiskCalculationParamContainer == null) {
-            presenceTracingRiskCalculationParamContainer =
-                configProvider.currentConfig.first().presenceTracing.riskCalculationParameters
-            Timber.d(presenceTracingRiskCalculationParamContainer.toString())
+        mutex.withLock {
```

Comment:
I think we could get compiler nullchecks from kotlins and return non-null here, can you try:

```kotlin
private suspend fun getRiskCalculationParameters(): PresenceTracingRiskCalculationParamContainer = mutex.withLock {
        presenceTracingRiskCalculationParamContainer.let {
            if (it == null) {
                val newParams = configProvider.currentConfig.first().presenceTracing.riskCalculationParameters
                Timber.d("New params %s", newParams)
                presenceTracingRiskCalculationParamContainer = newParams
                newParams
            } else {
                it
            }
        }
    }
```